test/integration/setup.js

```diff
@@ -1,128 +1,104 @@
-  const DisposableAssessment = artifacts.require('DisposableAssessment');
-  const DisposableYieldTokenIncidents = artifacts.require('DisposableYieldTokenIncidents');
-  const DisposableIndividualClaims = artifacts.require('DisposableIndividualClaims');
-  const DisposableCover = artifacts.require('DisposableCover');
-
-  // target contracts
-  const NXMaster = artifacts.require('NXMaster');
-  const MemberRoles = artifacts.require('MemberRoles');
-  const TokenController = artifacts.require('TokenController');
-  const ProposalCategory = artifacts.require('ProposalCategory');
-  const Governance = artifacts.require('Governance');
-  const PooledStaking = artifacts.require('LegacyPooledStaking');
-  const Gateway = artifacts.require('LegacyGateway');
-  const YieldTokenIncidents = artifacts.require('YieldTokenIncidents');
-  const IndividualClaims = artifacts.require('IndividualClaims');
-  const Assessment = artifacts.require('Assessment');
+  const NXMToken = await ethers.getContractFactory('NXMToken');
+  // const LegacyClaims = await ethers.getContractFactory('LegacyClaims');
+  // const LegacyIncidents = await ethers.getContractFactory('LegacyIncidents');
+  // const LegacyClaimsData = await ethers.getContractFactory('LegacyClaimsData');
+  const LegacyClaimsReward = await ethers.getContractFactory('LegacyClaimsReward');
+  const DisposableMCR = await ethers.getContractFactory('DisposableMCR');
+  const MCR = await ethers.getContractFactory('MCR');
+  const Pool = await ethers.getContractFactory('Pool');
+  const QuotationData = await ethers.getContractFactory('LegacyQuotationData');
+  const PriceFeedOracle = await ethers.getContractFactory('PriceFeedOracle');
+  const SwapOperator = await ethers.getContractFactory('CowSwapOperator');
+  const CoverNFT = await ethers.getContractFactory('CoverNFT');
+  // const StakingPool = await ethers.getContractFactory('StakingPool');
+
+  const CoverUtilsLib = await ethers.getContractFactory('CoverUtilsLib');
 
   const signers = await ethers.getSigners();
   const ethersAccounts = getAccounts(signers);
 
   // external
-  const WETH9 = artifacts.require('WETH9');
-  const CSMockSettlement = artifacts.require('CSMockSettlement');
-  const CSMockVaultRelayer = artifacts.require('CSMockVaultRelayer');
+  const WETH9 = await ethers.getContractFactory('WETH9');
+  const CSMockSettlement = await ethers.getContractFactory('CSMockSettlement');
+  const CSMockVaultRelayer = await ethers.getContractFactory('CSMockVaultRelayer');
 
-  const coverUtilsLib = await CoverUtilsLib.new();
-  DisposableCover.link(coverUtilsLib);
+  const deployImmutable = async (contractFactory, constructorArgs = []) => {
+    const instance = await contractFactory.deploy(...constructorArgs);
 
-  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
-  const QE = '0x51042c4d8936a7764d18370a6a0762b860bb8e07';
-  const INITIAL_SUPPLY = ether('15000000000');
+    return instance;
+  };
```

Comment:
The `deploy.js` script needed the `deployImmutable` function because it collected the artifacts for contract verification. However here it doesn't seem to have add any value and it's not used consistently - in some places it's used, in others `<ContractFactory>.deploy` is used. Unless I'm missing something, let's remove this one entirely.